client/pages/register.jsx

```diff
@@ -167,181 +141,349 @@ export default function RegisterPage(props) {
+                  <Col xs="3" align="right" className=" vertAlign textField">
+                    username
+                  </Col>
+                  <Col xs="9">
+                    <Form>
+                      <FormGroup>
+                        <Input
+                          type="username"
+                          name="username"
+                          minLength="8"
+                          maxLength="64"
+                          value={username}
+                          onChange={e => setUsername(e.target.value)}
+                          required
+                        />
+                      </FormGroup>
+                    </Form>
+                  </Col>
+                </Row>
+                {/* email */}
+                <Row align="middle" justify="center">
+                  <Col xs="3" align="right" className=" vertAlign textField">
+                    email
+                  </Col>
+                  <Col xs="9">
+                    <Form>
+                      <FormGroup>
+                        <Input
+                          type="email"
+                          name="email"
+                          id="exampleEmail"
+                          maxLength="64"
+                          pattern={EMAIL_REGEX}
+                          value={email}
+                          onChange={e => setEmail(e.target.value)}
+                          required
+                        />
+                      </FormGroup>
+                    </Form>
+                  </Col>
+                </Row>
+                {/* password */}
+                <Row>
+                  <Col xs="3" align="right" className=" vertAlign textField">
+                    password
+                  </Col>
+                  <Col xs="9">
+                    <Form>
+                      <FormGroup>
+                        <Input
+                          type="password"
+                          name="password"
+                          minLength="8"
+                          maxLength="64"
+                          value={password}
+                          onChange={e => setPassword(e.target.value)}
+                          required
+                        />
+                      </FormGroup>
+                    </Form>
+                  </Col>
+                </Row>
+
+                {/* confirm password */}
+                <Row>
+                  <Col xs="3" align="right" className=" vertAlign textField">
+                    confirm password
+                  </Col>
+                  <Col xs="9">
+                    <Form>
+                      <FormGroup>
+                        <Input
+                          type="password"
+                          name="password2"
+                          minLength="8"
+                          maxLength="64"
+                          value={password2}
+                          onChange={e => setPassword2(e.target.value)}
```

Comment:
I think we can just do `{setPassword2}` here. Correct me if I'm wrong though. There's a few places you would need to change if thats the case.